utils/paths.go

```diff
@@ -35,10 +38,24 @@ func GetCurationCacheFolder() (string, error) {
 	return filepath.Join(curationFolder, "cache"), nil
 }
 
-func GetCurationMavenCacheFolder() (string, error) {
+func GetCurationMavenCacheFolder(withProjectDir bool) (string, error) {
 	curationFolder, err := GetCurationCacheFolder()
 	if err != nil {
 		return "", err
 	}
-	return filepath.Join(curationFolder, "maven"), nil
+	projectDir := ""
+	if withProjectDir {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		// #nosec G401 -- Not a secret hash.
+		hasher := sha1.New()
+		_, err = hasher.Write([]byte(workingDir))
+		if err != nil {
+			return "", err
+		}
+		projectDir = hex.EncodeToString(hasher.Sum(nil))
+	}
+	return filepath.Join(curationFolder, "maven", projectDir), nil
```

Comment:
```suggestion
        workingDir, err := os.Getwd()
	if err != nil {
		return
	}
	// #nosec G401 -- Not a secret hash.
	hasher := sha1.New()
	if _, err = hasher.Write([]byte(workingDir)); err != nil {
		return
	}
	projectDir = filepath.Join(curationFolder, "maven", hex.EncodeToString(hasher.Sum(nil)))
	return
```
change func return signature to `(projectDir string, err errror)`